prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -594,31 +581,68 @@ class ReportFunction : Logging {
                 return sb.toString()
             }
 
-            fun writeConsolidatedArray(field: String, array: List<ResultDetail>) {
-                val rowsByGroupingId = hashMapOf<String, MutableList<Int>>()
+            fun createPropertiesByGroupingId(array: List<ResultDetail>): List<HashMap<String, out Any>> {
```

Comment:
List<HashMap<String, out Any>> is confusing - Turn this into a data class that has the four Maps in it, as named fields in the data class.

Each of the four objects can use the abstract class Map, even though they are created using hashMapOf.

Reply:
Thanks for the advice, Jim! Check the most recent updates and let me know what you think 👍🏼